plaso/parsers/nids_fast.py

```diff
@@ -0,0 +1,327 @@
+
+
+class NIDSFastParser(tp.PyparsingSingleLineTextParser):
+  """NIDS alert data parser for fast format (alert_fast.txt and fast.log)."""
+
+  NAME = 'nids:alert:fast'
+  DATA_FORMAT = 'NIDS fast alert log'
+
+  _FASTLOG_VERIFICATION_PATTERN = (
+      # Date regex
+      r'^(\d{2}\/)?\d{2}\/\d{2}\-\d{2}:\d{2}:\d{2}.\d{6}\s*'
+      # Separator ([**])
+      r'\[\*\*\]\s*'
+      # Rule identifier
+      r'\[\d*:\d*:\d*\]\s*'
+      # Message
+      r'"?.*\"?\s*\[\*\*\]\s*'
+      # Optional Classification
+      r'(\[Classification:\s.*\])?\s*'
+      # Optional Priority
+      r'(\[Priority\:\s*\d{1}\])?\s*'
+      # Procotol
+      r'\{\w+\}\s*'
+      # Source IPv6 address
+      r'((([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:'
+      r'|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a -fA-F]{1,4}:){1,5}'
+      r'(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[ 0-9a-fA-F]{1,4})'
+      r'{1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4} |([0-9a-fA-F]'
+      r'{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((: [0-9a-fA-F]'
+      r'{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4})'
+      r'{0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2 [0-4]'
+      r'|1{0,1}[0-9]){0,1}[0-9]).){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}'
+      r'[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}'
+      r'[0-9]).){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))'
+      # Source IPv4 address
+      r'|(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?'
+      r'[0-9][0-9]?)\.(25[0-5] |2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|'
+      r'2[0-4][0-9]|[01]?[0-9][0-9]?))'
+      # Optional source port
+      r'(:\d*)?\s*'
+      # Separator '->'
+      r'\-\>\s*'
+      # Destination IPv6-address
+      r'((([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:'
+      r'|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a -fA-F]{1,4}:){1,5}'
+      r'(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[ 0-9a-fA-F]{1,4})'
+      r'{1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4} |([0-9a-fA-F]'
+      r'{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((: [0-9a-fA-F]'
+      r'{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4})'
+      r'{0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2 [0-4]'
+      r'|1{0,1}[0-9]){0,1}[0-9]).){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}'
+      r'[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}'
+      r'[0-9]).){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))'
+      # Destination IPv4 address
+      r'|(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?'
+      r'[0-9][0-9]?)\.(25[0-5] |2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|'
+      r'2[0-4][0-9]|[01]?[0-9][0-9]?))'
+      # Optional destination port
+      r'(:\d*)?$'
+  )
+  _VERIFICATION_REGEX = re.compile(_FASTLOG_VERIFICATION_PATTERN)
+
+  _PYPARSING_COMPONENTS = {
```

Comment:
The double indirection is not needed IMHO

Reply:
Could you explain what you want to be changed here exactly?

Reply:
no need to make changes, I'm making some also for the recent text plugin change
but what I mean there is no need to use a dict like _PYPARSING_COMPONENTS and then doing another lookup to get the pyparsing definition, I opt to use static class definitions.